pkg/queue/stats.go

```diff
@@ -102,27 +118,35 @@ func NewStats(podName string, channels Channels, startedAt time.Time) *Stats {
 					totalTimeUsed += val
 				}
 
-				var avg float64
+				var avg, avgp float64
 				if totalTimeUsed > 0 {
 					for c, val := range timeOnConcurrency {
 						ratio := float64(val) / float64(totalTimeUsed)
 						avg += float64(c) * ratio
 					}
+					for c, val := range timeOnProxiedCon {
+						ratio := float64(val) / float64(totalTimeUsed)
+						avgp += float64(c) * ratio
+					}
```

Comment:
Should we make this a function to prevent the duplication?

Like

```golang
func averageConcurrency(times map[int32]time.Duration, totalTimeUsed Duration) float64 {
    if totalTimeUsed == 0 {
        return 0
    }
    for c, val := range times {
        ratio := float64(val) / float64(totalTimeUsed)
        avgp += float64(c) * ratio
    }
}
```

Reply:
done, thanks.

Reply:
I'd have pulled the `if totalTimeUsed` into the function as shown below and pulled it out of the aggregation logic for better readability. That's a nit though, up to you to fix or not.